docs/source/user-guide/supported_models.rst

```diff
@@ -0,0 +1,56 @@
+.. _supported_models:
+
+====================
+Supported Models
+====================
+
+Deepchecks supports a minmal version of the sklearn api.
+
```

Comment:
This whole part should be much more elaborate...
a sentence about motivation (what happens if you don't supply us with a compatible model? then we can't run all checks that require predictions - e.g. all performance related etc...)
Examples of common models that support the scikit-learn api (all scikit learn models obviously, but also other models that we suppport as-is... this should be the first thing - so the user quickly understands "what are the straight forward things we support easily", without needing to go into the nitty-gritty of the API etc.)

Reply:
+1

Reply:
Hi, I see you updated here, but I still don't feel this is "user facing".
The user comes here to try and understand:
Is my model supported?
The first paragraph should lead him as quickly as possible to this answer...